iceoryx_examples/waitset/README.md

```diff
@@ -1,6 +1,566 @@
+group has new samples.
+
+### Individual
+When every _Triggerable_ requires a different reaction we need to know the 
+origin of a _Trigger_. We can call `trigger.doesOriginateFrom(TriggerOrigin)`
+which will return true if the trigger originates from _TriggerOrigin_ and
+otherwise false.
+
+We start this example by creating a _WaitSet_ and attaching the `shutdownTrigger`
+to handle `CTRL-c`.
+```cpp
+iox::popo::WaitSet waitset;
+
+shutdownTrigger.attachToWaitset(waitset);
+```
+
+Additionally, we create two subscriber, subscribe them to our service and attach
+them to the waitset to let them inform us whenever they receive a new sample.
+```cpp
+iox::popo::TypedSubscriber<CounterTopic> subscriber1({"Radar", "FrontLeft", "Counter"});
+iox::popo::TypedSubscriber<CounterTopic> subscriber2({"Radar", "FrontLeft", "Counter"});
+
+subscriber1.subscribe();
+subscriber2.subscribe();
+
+subscriber1.attachToWaitset(waitset, iox::popo::SubscriberEvent::HAS_NEW_SAMPLES);
+subscriber2.attachToWaitset(waitset, iox::popo::SubscriberEvent::HAS_NEW_SAMPLES);
+```
+
+With that set up we enter the event loop and handle the program termination
+first.
+```cpp
+while (true)
+{
+    auto triggerVector = waitset.wait();
+
+    for (auto& trigger : triggerVector)
+    {
+        if (trigger.doesOriginateFrom(&shutdownTrigger))
+        {
+            return (EXIT_SUCCESS);
+        }
+```
+
+If `subscriber1` we would like to state that subscriber 1 has received the 
+following number X. But for `subscriber2` we just dismiss the received samples.
+We accomplish this by asking the `trigger` if it originated from the 
+corresponding subscriber. If so we act.
+```cpp
+        else if (trigger.doesOriginateFrom(&subscriber1))
+        {
+            subscriber1.take().and_then([&](iox::popo::Sample<const CounterTopic>& sample) {
+                std::cout << " subscriber 1 received: " << sample->counter << std::endl;
+            });
+        }
+        if (trigger.doesOriginateFrom(&subscriber2))
+        {
+            subscriber2.releaseQueuedSamples();
+            std::cout << "subscriber 2 received something - dont care\n";
+        }
+```
+
+### Sync
+Lets say we have `SomeClass` and would like to execute a cyclic static call 
+`cyclicRun`
+in that class every second. We could execute any arbitrary algorithm in there
+but for now we just print `activation callback`. The class could look like
+```cpp
+class SomeClass
+{
+  public:
+    static void cyclicRun(iox::popo::UserTrigger*)
+    {
+        std::cout << "activation callback\n";
+        trigger->resetTrigger();
+    }
+};
+```
+**Important** We need to reset the trigger otherwise the WaitSet would notify
+us immediately again since it is state based.
+
+We begin as always, by creating a _WaitSet_ and attaching the `shutdownTrigger` to 
+it.
+```cpp
+iox::popo::WaitSet waitset;
+
+// attach shutdownTrigger to handle CTRL+C
+shutdownTrigger.attachToWaitset(waitset);
```

Comment:
Which triggerID is used when noone is set?

Reply:
explained and documented